src/ifaddrs.rs

```diff
@@ -0,0 +1,180 @@
+//! Query network interface addresses
+//!
+//! Uses the Linux and/or BSD specific function getifaddrs to query the list
+//! of interfaces and their associated addresses.
+
+use std::ffi;
+use std::iter::Iterator;
+use std::mem;
+use std::option::Option;
+
+use libc;
+use {Result, Error};
+
+use sys::socket::InetAddr;
+
+#[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "fuchsia"))]
+mod flags {
+    use libc;
+    libc_bitflags!(
+        /// Flags for `InterfaceAddress` entries returned by `getifaddrs`
```

Comment:
That comment doesn't really provide any detail. Can you summarize what these values actually refer to?